tests/functional/object/mcg/ui/test_bucket_policy_ui.py

```diff
@@ -0,0 +1,232 @@
+            },
+            {
+                "name": "AllowReadWriteAccessToFolder",
+                "method": "set_bucket_policy_folder_access_ui",
+                "params": {
+                    "folder_path": "documents",
+                    "account_list": ["123456789012"],
+                },
+            },
+        ],
+    )
+    def test_set_bucket_policy_ui(self, setup_ui_class_factory, policy_config):
+        """
+        Test setting various bucket policies via UI.
+
+        This test follows the workflow:
+        1. Navigate to Object Storage
+        2. Create OBC bucket if needed for account-specific policies
+        3. Click on appropriate bucket
+        4. Go to Permissions tab
+        5. Click "Start from scratch" to activate policy editor
+        6. Generate policy JSON programmatically
+        7. Set the policy JSON in the code editor
+        8. Apply the policy
+        9. Confirm in modal
+
+        Args:
+            policy_config (dict): Configuration containing policy name, method, and parameters
+
+        Raises:
+            pytest.skip: If no buckets are available for testing
+        """
+        setup_ui_class_factory()
+        policy_name = policy_config["name"]
+        method_name = policy_config["method"]
+        params = policy_config["params"]
+
+        logger.info(f"Starting test to set {policy_name} bucket policy")
+
+        bucket_ui = BucketsTab()
+        bucket_ui.navigate_buckets_page()
+
+        # Check if this policy requires account ID (OBC bucket)
+        account_dependent_policies = [
+            "AllowAccessToSpecificAccount",
+            "AllowReadWriteAccessToFolder",
+        ]
+
+        target_bucket_name = None
+
+        if policy_name in account_dependent_policies:
+            # Get current bucket list to identify new bucket after creation
+            initial_buckets = set(bucket_ui.get_buckets_list())
+
+            # Create OBC bucket for account-dependent policies
+            bucket_ui.create_bucket_ui(method="obc")
+
+            # Navigate back to buckets page and wait for new bucket to appear
+            bucket_ui.navigate_buckets_page()
+            bucket_ui.page_has_loaded(sleep_time=2)
+
+            # Wait for new OBC bucket to appear (up to 30 seconds)
+            target_bucket_name = None
+            for attempt in range(6):  # 6 attempts * 5 seconds = 30 seconds max wait
+                current_buckets = set(bucket_ui.get_buckets_list())
+                new_buckets = current_buckets - initial_buckets
+
+                # Look for newly created bucket with OBC pattern
+                obc_buckets = [
+                    b for b in new_buckets if b.startswith("test-bucket-obc-")
+                ]
+
+                if obc_buckets:
+                    target_bucket_name = obc_buckets[0]  # Use first OBC bucket found
+                    break
+
+                if attempt < 5:  # Don't sleep on last attempt
+                    time.sleep(5)
```

Comment:
You can use TimeoutSampler for this:
```suggestion
            for current_buckets_sample in TimeoutSampler(
                timeout=30, sleep=5, func=bucket_ui.get_buckets_list
             ):
                 new_buckets = set(current_buckets_sample) - initial_buckets
                 
                 # Look for newly created bucket with OBC pattern
                 obc_buckets = [
                    b for b in new_buckets if b.startswith("test-bucket-obc-")
                 ]
                 
                 if obc_buckets:
                     target_bucket_name = obc_buckets[0]
                     break
```